cvat/apps/engine/serializers.py

```diff
@@ -405,17 +405,76 @@ def update(self, instance, validated_data):
             instance.bug_tracker)
         instance.subset = validated_data.get('subset', instance.subset)
         labels = validated_data.get('label_set', [])
-        for label in labels:
-            LabelSerializer.update_instance(label, instance)
+        if instance.project_id is None:
+            for label in labels:
+                LabelSerializer.update_instance(label, instance)
+        validated_project_id = validated_data.get('project_id', None)
+        if validated_project_id is not None and validated_project_id != instance.project_id:
+            project = models.Project.objects.get(id=validated_data.get('project_id', None))
+            if instance.project_id is None:
+                for old_label in instance.label_set.all():
+                    try:
+                        new_label = project.label_set.filter(name=old_label.name).first()
```

Comment:
For every label it is a separate DB call. Probably it isn't a problem for now. Let's keep it as is.

Reply:
In that case, there are already 4 DB calls for every label at least. I will try to improve it.